test/integration/runtime_int_test.go

```diff
@@ -212,6 +212,36 @@ func (suite *RuntimeIntegrationTestSuite) TestIgnoreEnvironmentVars() {
 	cp.ExpectExitCode(0)
 }
 
+func (suite *RuntimeIntegrationTestSuite) TestRuntimeCache() {
+	suite.OnlyRunForTags(tagsuite.Critical)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	ts.PrepareEmptyProject()
+
+	cp := ts.Spawn("install", "shared/zlib")
+	cp.Expect("Downloading")
+	cp.ExpectExitCode(0)
+
+	cp = ts.Spawn("switch", "mingw") // should not remove cached shared/zlib artifact
+	cp.ExpectExitCode(0)
+
+	cp = ts.Spawn("install", "shared/zlib")
+	cp.Expect("Installing")
+	cp.ExpectExitCode(0)
+	suite.Assert().NotContains(cp.Snapshot(), "Downloading", "shared/zlib should have been cached")
```

Comment:
In addition to this it would be nice if we could also assert that the artifact is still in the depot. Because the "Downloading" contract is very implicit. eg. we could rename "Downloading" to "Obtaining" and not think to touch this test, and it would still be passing.